src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -232,6 +244,15 @@ public override Task SetVariableAsync(string name, object value)
             return Task.CompletedTask;
         }
 
+        public static Dictionary<String, Object> GetPSObjectProperties(PSObject psObject){
```

Comment:
Please don't add the utility method in `PowerShellKernel.cs`. Instead, add it to `PowerShellExtensions.cs` as an extension method.
Also, move all the logic that deal with PSObject in the extension method, not just the logic that handles `PSCustomObject`.
Example:
```c#
public static object Unwrap(this PSObject psobj)
{
    object obj = psObj.BaseObject;
    if (obj is PSCustomObject)
    {
        ....
        obj = table;
    }

    return obj;
}
```

In `out-display`, use it like this (similarly in `TryGetVariable`):

```c#
object obj = InputObject is PSObject psObj ? psObj.Unwrap() : InputObject;
DisplayedValue displayedValue = display(obj, MimeType);
```